app/models/user.rb

```diff
@@ -10,6 +10,15 @@ class User < ActiveRecord::Base
   belongs_to :craftsman
   before_create :associate_craftsman
 
+  validate :password_complexity
+  def password_complexity
+    if password.present?
+      if !password.match(/^(?=.*[a-z])(?=.*[A-Z])/)
+        errors.add :password, "Password complexity requirement not met"
```

Comment:
1) The signup page displays this as "Password Password complexity requirement not met" with the flash message, so you can chop off the "Password" here

2) Can the message tell the user what the complexity requirements actually are? So they can see what they need to include and actually do it. It'll help us too because I can't even read the regex myself lol

Reply:
okay how's this? (also made it more robust once I figured out what it actually did haha)

Reply:
Can we include a couple more special characters in the regex like `- + ; : ' "` and maybe square/curly brackets?

I kept trying to make a user with `Password1-` and was so confused why it wasn't working

Reply:
Sure, want to just add that yourself in a commit? I won't be able to edit it until later tonight. My go to is "P@ssw0rd" haha

Reply:
oh nice, and yeah sure 👍